samtranslator/model/eventsources/pull.py

```diff
@@ -283,3 +290,135 @@ def get_policy_statements(self):
+                self.relative_id,
+                "No Topics provided for self managed kafka as an event source",
+            )
+
+        if not (len(self.Topics) == 1):
+            raise InvalidEventException(
+                self.relative_id,
+                "Topics for self managed kafka only supports single configuration entry.",
+            )
+
+        if not self.SourceAccessConfigurations:
+            raise InvalidEventException(
+                self.relative_id,
+                "No SourceAccessConfigurations for self managed kafka event provided.",
+            )
+        document = self.generate_policy_document()
+        return [document]
+
+    def generate_policy_document(self):
+        document = {
+            "PolicyDocument": {
+                "Statement": [],
+                "Version": "2012-10-17",
+            },
+            "PolicyName": "SelfManagedKafkaExecutionRolePolicy",
+        }
+
+        authentication_uri, has_vpc_config = self.get_secret_key()
+        if authentication_uri:
+            secret_manager = self.get_secret_manger_secret(authentication_uri)
+            document["PolicyDocument"]["Statement"].append(secret_manager)
+
+        if has_vpc_config:
+            vpc_permissions = self.get_vpc_permission()
+            document["PolicyDocument"]["Statement"].append(vpc_permissions)
+
+        if self.SecretsManagerKmsKeyId:
+            kms_policy = self.get_kms_policy()
+            document["PolicyDocument"]["Statement"].append(kms_policy)
+
+        return document
+
+    def get_secret_key(self):
+        authentication_uri = None
+        has_vpc_subnet = False
+        has_vpc_security_group = False
+        for config in self.SourceAccessConfigurations:
+            if config["Type"] == "VPC_SUBNET":
+                if not config["URI"]:
+                    raise InvalidEventException(
+                        self.relative_id,
+                        "No VPC_SUBNET URI property specified in SourceAccessConfigurations for self managed kafka event.",
+                    )
+                has_vpc_subnet = True
+
+            if config["Type"] == "VPC_SECURITY_GROUP":
+                if not config["URI"]:
+                    raise InvalidEventException(
+                        self.relative_id,
+                        "No VPC_SECURITY_GROUP URI property specified in SourceAccessConfigurations for self managed kafka event.",
+                    )
+                has_vpc_security_group = True
+
+            if config["Type"] in self.AUTH_MECHANISM:
+                if authentication_uri:
+                    raise InvalidEventException(
+                        self.relative_id,
+                        "Multiple auth mechanism properties specified in SourceAccessConfigurations for self managed kafka event.",
+                    )
+                authentication_uri = config["URI"]
+
+        if not (has_vpc_subnet == has_vpc_security_group):
```

Comment:
For readability, maybe change:
- The code to `(not has_vpc_subnet && has_vpc_security_group) or (has_vpc_subnet and not has_vpc_security_group)`
- The error message to: `VPC_SUBNET and VPC_SECURITY_GROUP in SourceAccessConfigurations for SelfManagedKafka must be both provided.`

Do we also need to throw an error if `has_vpc_subnet` is True and `has_vpc_security_group` is True and `authentication_uri` is not empty?

Last thought, for `self.AUTH_MECHANISM`, we don't check that `config["URI"]` has a value, we just assign it to `authentication_uri`. Shall we?

Reply:
> Do we also need to throw an error if `has_vpc_subnet` is True and `has_vpc_security_group` is True and `authentication_uri` is not empty?

No they are independent, either the customer provides just the auth_mechanism or the VPC credentials or both.